R/use_extendr.R

```diff
@@ -23,11 +23,17 @@
 use_extendr <- function(path = ".",
                         crate_name = NULL,
                         lib_name = NULL,
-                        quiet = getOption("usethis.quiet", FALSE),
+                        quiet = FALSE,
                         edition = c("2021", "2018")) {
-  usethis_quiet <- getOption("usethis.quiet")
-  on.exit(options(usethis.quiet = usethis_quiet))
-  options(usethis.quiet = quiet)
+
+  # https://github.com/r-lib/cli/issues/434
+
+  if (quiet) {
+    old_cli_handler <- options("cli.default_handler")
+    options("cli.default_handler" = function(...) { })
+    on.exit(options("cli.default_handler" = old_cli_handler))
```

Comment:
I think we can use `withr::local_options` here.

https://github.com/extendr/rextendr/blob/232133b32125d2f82748668f6cabe64cddb0a1b5/R/use_extendr.R#L205

Reply:
I think we can! Was unaware of `local_options()`. Will adjust.

Reply:
Addressed this. Also modified other instances of usethis.quiet to utilize cli options since `usethis` is not used for messaging--i.e. no `ui_*()` functions are being called.